datafusion/physical-expr/src/physical_expr.rs

```diff
@@ -109,8 +113,8 @@ fn scatter(mask: &BooleanArray, truthy: &dyn Array) -> Result<ArrayRef> {
         filled = end;
     });
     // the remaining part is falsy
-    if filled < truthy.len() {
-        mutable.extend_nulls(truthy.len() - filled);
+    if filled < mask.len() {
```

Comment:
this is a bug I found while writing some more tests (namely that the output is supposed to be the size of `mask` not the size of `truthy` I think

Reply:
Yes! And it's not covered by test util this PR.